apps/heft/src/plugins/ApiExtractorPlugin/ApiExtractorRunner.ts

```diff
@@ -0,0 +1,144 @@
+        switch (message.logLevel) {
+          case apiExtractor.ExtractorLogLevel.Error: {
+            let logMessage: (IColorableSequence | string)[];
+            if (message.sourceFilePath) {
+              const buildFolderRelativeFilePath: string = path.relative(
+                this._configuration.buildFolder,
+                message.sourceFilePath
+              );
+              logMessage = [
+                Colors.red(
+                  `${buildFolderRelativeFilePath}(${message.sourceFileLine},${message.sourceFileColumn})`
+                ),
+                Colors.white(' - '),
+                Colors.gray(`(${message.category})`),
+                ' ',
+                Colors.red(message.text)
+              ];
+            } else {
+              logMessage = [message.text];
+            }
+
+            this._terminal.writeErrorLine(...logMessage);
+            break;
+          }
+
+          case apiExtractor.ExtractorLogLevel.Warning: {
+            let logMessage: (IColorableSequence | string)[];
+            if (message.sourceFilePath) {
+              const buildFolderRelativeFilePath: string = path.relative(
+                this._configuration.buildFolder,
+                message.sourceFilePath
+              );
+              logMessage = [
+                Colors.yellow(
+                  `${buildFolderRelativeFilePath}(${message.sourceFileLine},${message.sourceFileColumn})`
+                ),
+                Colors.white(' - '),
+                Colors.gray(`(${message.category})`),
+                ' ',
+                Colors.yellow(message.text)
+              ];
+            } else {
+              logMessage = [message.text];
+            }
+
+            this._terminal.writeWarningLine(...logMessage);
+            break;
+          }
+
+          case apiExtractor.ExtractorLogLevel.Verbose: {
+            this._terminal.writeVerboseLine(message.text);
+            break;
+          }
+
+          case apiExtractor.ExtractorLogLevel.None:
+          case apiExtractor.ExtractorLogLevel.Info: {
+            this._terminal.writeLine(message.text);
+            break;
+          }
+
+          default:
+            throw new Error(`Unexpected API Extractor log level: ${message.logLevel}`);
+        }
+
+        message.handled = true;
+      }
+    };
+
+    this._terminal.writeLine(`Using API Extractor version ${apiExtractor.Extractor.version}`);
+
+    const apiExtractorResult: TApiExtractor.ExtractorResult = apiExtractor.Extractor.invoke(
+      extractorConfig,
+      extractorOptions
+    );
+
+    const { errorCount, warningCount } = apiExtractorResult;
+    if (errorCount > 0) {
+      this._terminal.writeErrorLine(
+        `Encountered ${errorCount} API Extractor error${errorCount > 1 ? 's' : ''}`
+      );
+    }
+
+    if (warningCount > 0) {
+      this._terminal.writeErrorLine(
+        `Encountered ${warningCount} API Extractor warning${warningCount > 1 ? 's' : ''}`
```

Comment:
Seems slightly odd to write a log message for both errors and warnings. The real AE does this:
```
      if (extractorResult.errorCount > 0) {
        console.log(os.EOL + colors.red('API Extractor completed with errors'));
      } else {
        console.log(os.EOL + colors.yellow('API Extractor completed with warnings'));
      }
```